src/vm/methodtablebuilder.cpp

```diff
@@ -3828,6 +3829,20 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
             }
         }
 
+        // Determine if a field is a fixed buffer.
+        {
```

Comment:
What is the performance impact of fetching a custom attribute on every field in the type loader now?

Reply:
I'll try to go run some benchmarks.

I also just went and checked with roslyn where fixed buffers can appear and it looks like they can only be in value types. I'll go add in some conditions here to only look for the attribute when absolutely necessary to reduce the impact.

Reply:
And after a few more runs of those benchmarks again in a different order, I've now seen all of the benchmarks I've run come in at 0% perf hit or 1% perf improvement (which is very likely just noise).